cmd/dagger-graph/go.mod

```diff
@@ -0,0 +1,73 @@
+module github.com/dagger/dagger/cmd/dagger-graph
```

Comment:
Is the separate go.mod necessary?

Reply:
I didn't want to pollute our main go.mod with `d2`, happy to change though

Reply:
Since go modules are lazily loaded I don't see a huge downside to keeping a single go.mo and a slight upside in that it makes it easier to maintain if we want this code to have a dependency on the main engine codebase so it can use the structs in `core/pipeline.go`.

Not a huge deal either way though

Reply:
Pushed this in a separate commit. It's not too bad actually, looks ok to you?

Reply:
Yep, LGTM